examples/plotting/server/animated_fourier.py

```diff
@@ -0,0 +1,195 @@
+        cx, cy = cp['x'][0], cp['y'][0]
+
+        if i == len(foos)-1:
+            sources[i]['floating_point'].data['x'] = [shift]
+            sources[i]['floating_point'].data['y'] = [cy]
+
+
+def update_centric_sources(sources, foos, newx, ind, cfoos):
+    for i, foo in enumerate(foos):
+        get_new_sources(newx, foo, sources[i], cfoos[i])
+
+
+def create_centric_plot(foos, title='', r = 1, y_range=None, period = np.pi/2., cfoos=None):
+    if y_range is None:
+        y_range=[-2, 2]
+
+    p = figure(title=title, width=800, height=300, x_range=[-1.5, 10.5], y_range=y_range)
+    p.xgrid.bounds = (-2, 2)
+    p.xaxis.bounds = (-2, 2)
+
+    _sources = []
+    for i, foo in enumerate(foos):
+        sources = new_source()
+        get_new_sources(x, foo, sources, cfoos[i])
+        _sources.append(sources)
+
+        if i:
+            legend = "4sin(%(c)sx)/%(c)spi" % {'c': i*2+1}
+        else:
+            legend = "4sin(x)/pi"
+
+        p.line('base_x','y', color=palette[i], line_width=2, source=sources['curve'])
+        p.line('line_x', 'line_y', color=palette[i], line_width=2,
+                source=sources['lines'], legend=legend)
+
+        create_circle_glyphs(p, palette[i], sources)
+
+    return p, _sources
+
+# Create the series partials..
+# NOTE: We could create those dinamically but leaving as
+#       is improves readability
+f1 = lambda x: (4*np.sin(x))/np.pi
+f2 = lambda x: (4*np.sin(3*x))/(3*np.pi)
+f3 = lambda x: (4*np.sin(5*x))/(5*np.pi)
+f4 = lambda x: (4*np.sin(7*x))/(7*np.pi)
+cf1 = lambda x: (4*np.cos(x))/np.pi
+cf2 = lambda x: (4*np.cos(3*x))/(3*np.pi)
+cf3 = lambda x: (4*np.cos(5*x))/(5*np.pi)
+cf4 = lambda x: (4*np.cos(7*x))/(7*np.pi)
+fourier = OrderedDict(
+    fourier_4 = {
+        'f': lambda x: f1(x) + f2(x) + f3(x) + f4(x),
+        'fs': [f1, f2, f3, f4],
+        'cfs': [cf1, cf2, cf3, cf4]},
+)
+
+for k, p in fourier.items():
+    p['plot'], p['sources'] = create_plot(
+        p['fs'], 'Fourier (Sum of the first 4 Harmonic Circles)', r = p['f'](period), cfoos = p['cfs']
+    )
+
+for k, p in fourier.items():
+    p['cplot'], p['csources'] = create_centric_plot(
+        p['fs'], 'Fourier First 4 Harmonics & Harmonic Circles', r = p['f'](period), cfoos = p['cfs']
+    )
+
+# Open a session which will keep our local doc in sync with server
+session = push_session(curdoc())
+# Open the session in a browser
+layout = vplot(*[f['plot'] for f in fourier.values()] + [f['cplot'] for f in fourier.values()])
+session.show(layout)
+
+gind = 0
+def cb():
+    global gind
+    global newx
+    oldx = np.delete(newx, 0)
+    newx = np.hstack([oldx, [oldx[-1] + 2*np.pi/100]])
+    gind += 1
+
+    for k, p in fourier.items():
+        update_sources(p['sources'], p['fs'], newx, gind, p['cfs'])
+        update_centric_sources(p['csources'], p['fs'], newx, gind, p['cfs'])
+
+    if gind >= 99:
+        gind = 0
+
+# Add the callback function to bokeh server session
+curdoc().add_periodic_callback(cb, .05)
```

Comment:
This is an awesome example, but the purpose of it is to demonstrate the periodic callback. To that end, maybe we could have a more emphatic comment here.